src/main/java/org/folio/support/ExpirationTool.java

```diff
@@ -42,44 +46,67 @@ private ExpirationTool() {
     return future.compose(rs -> CompositeFuture.all(rs.getRows()
       .stream()
       .map(row -> doRequestExpirationForTenant(vertx, getTenant(row.getString("nspname"))))
-      .collect(Collectors.toList())));
+      .collect(Collectors.toList()))
+      .map(all -> null));
   }
 
-  private static Future<CompositeFuture> doRequestExpirationForTenant(Vertx vertx, String tenant) {
+  private static Future<Void> doRequestExpirationForTenant(Vertx vertx, String tenant) {
+
+    Future<Void> future = Future.future();
+
+    PostgresClient pgClient = PostgresClient.getInstance(vertx, tenant);
 
-    return getExpiredRequests(vertx, tenant)
-      .compose(requests -> closeRequests(vertx, tenant, requests))
-      .compose(v -> getOpenRequests(vertx, tenant))
-      .compose(requests -> reorderRequests(vertx, tenant, requests));
+    pgClient.startTx(conn -> getExpiredRequests(conn, vertx, tenant)
+      .compose(requests -> closeRequests(conn, vertx, tenant, requests))
+      .compose(v -> getOpenRequests(conn, vertx, tenant))
+      .compose(requests -> reorderRequests(conn, vertx, tenant, requests))
+      .setHandler(v -> {
+        if (v.failed()) {
+          pgClient.rollbackTx(conn, done -> future.fail(v.cause()));
+        } else {
+          pgClient.endTx(conn, done -> future.complete());
+        }
+      }));
+
+    return future;
   }
 
-  private static Future<List<Request>> getExpiredRequests(Vertx vertx, String tenant) {
+  private static Future<List<Request>> getExpiredRequests(AsyncResult<SQLConnection> conn, Vertx vertx, String tenant) {
 
     SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
     df.setTimeZone(TimeZone.getTimeZone(ZoneOffset.UTC));
 
     PostgresClient pgClient = PostgresClient.getInstance(vertx, tenant);
-    Future<Results<Request>> future = Future.future();
+    Future<ResultSet> future = Future.future();
 
-    String where = String.format("WHERE " +
+    String where = format("WHERE " +
         "(jsonb->>'status' = '%1$s' AND jsonb->>'requestExpirationDate' < '%3$s') OR " +
         "(jsonb->>'status' = '%2$s' AND jsonb->>'holdShelfExpirationDate' < '%3$s')",
 
       OPEN_NOT_YET_FILLED.value(),
       OPEN_AWAITING_PICKUP.value(),
       df.format(new Date()));
 
-    pgClient.get(REQUEST_TABLE, Request.class, "jsonb", where, false, false, false, future.completer());
 
-    return future.map(Results::getResults);
+    String fullTableName = format("%s.%s", PostgresClient.convertToPsqlStandard(tenant), REQUEST_TABLE);
+    String query = format("SELECT jsonb FROM %s %s", fullTableName, where);
+    pgClient.select(conn, query, future);
```

Comment:
Is this because there isn’t a way to do a get whilst specifying a connection / transaction?

Reply:
Yes. There is no get method overloading with `where` parameter.

Reply:
Is it worth raising a change for the RAML Module Builder for this to be added in future?

Reply:
Probably having such overloading would be better, but I am ok with such approach.